backend/src/types.ts

```diff
@@ -219,3 +219,25 @@ export type BuildStatus = {
   status: BUILD_PHASE;
   timestamp?: string;
 };
+
+export enum NotebookStatus {
+  VALIDITING,
+  SUCCESS,
+  FAILED
+}
+
+export type Notebook = {
+  name: string;
+  repo: string;
+  description?: string;
+  status?: NotebookStatus;
+  user?: string;
+  uploaded?: Date;
+  visible?: boolean;
+  packages?: NotebookPackage[];
+}
```

Comment:
I also think we're missing here the field for messages -> this is populated in case the import finds some violations and refuses to import. It's a suggestion to user about what went wrong. See:

https://marvelapp.com/prototype/6dcfc3h/screen/85139862
![image](https://user-images.githubusercontent.com/7453394/159482427-f8527687-93bd-40dd-b9ab-06f1b5cefe87.png)

In the OpenShift resource it is defined as a JSON encoded string (we can parse it in the backend):

```json
[{"severity":"error","message":"Python version does not meet minimal required version ('3.8.0')"}]
```

Reply:
Wouldn't this come back as an error message in say 400 response.  e.g. https://github.com/dlabaj/odh-dashboard/blob/660f6a3b24caa63c5578d33dcb4975a54f066146/backend/src/routes/api/notebook/notebooksUtils.ts#L66 or not?

Reply:
I don't think so.. I think you want to get a 200 message and still receive the messages/recommendations.

The import action is an async process. So the API doesn't know this data at the time or a `POST` to `/api/notebooks`.

When you do `GET` on `/api/notebooks` you should receive all the images and some of them will have status "Succeeded" and some of them may have status "Failed" with the messages populated...

Presence of the messages doesn't mean the resource doesn't exist (or that you should not be requesting it - as 400 would imply), it means the import has failed due to any reason and you should receive a valid 200 with a payload + those details. Imagine this being a failing/crashing pod resource. If you do a get request for it, you still get the object json/yaml no matter the failing state...

Reply:
okay will add this field.